airbyte/_writers/file_writers.py

```diff
@@ -61,7 +62,11 @@ def _get_new_cache_file_path(
         batch_id = batch_id or str(ulid.ULID())
         target_dir = Path(self._cache_dir)
         target_dir.mkdir(parents=True, exist_ok=True)
-        return target_dir / f"{stream_name}_{batch_id}{self.default_cache_file_suffix}"
+        # If a stream contains a special Character, the temporary jsonl.gz
+        # file can't be created, because of OS restrictions. Therefore, we
+        # remove the special characters.
```

Comment:
```suggestion
        # If a stream contains a special Character, the temporary jsonl.gz
        # file can't be created, because of OS restrictions. Therefore, we
        # remove the special characters.
        # Specifically: we remove any of these characters: `<>:"/\|?*`
        # and we remove characters in the ASCII range from 0 to 31.
```